level_depth=(0, 0) is invalid

**Describe the bug**
The default value of `BaseRecursiveDriver ` is `level_depth=(0, 0)`, which is an invalid option. The traverse range is **right exclusive**.

**Describe how you solve it**
change it to `(0, 1)`

- - -
<!-- Optional, but really help us locate the problem faster -->

**Environment**
<!-- Run `jina --version-full` and copy paste the output here -->

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->



┆Issue is synchronized with this [Jira Task](https://jinaai.atlassian.net/browse/JINACORE-163) by [Unito](https://www.unito.io/learn-more)

This change leads to the failure of the following tests.

```
FAILED tests/unit/drivers/test_reduce_all_driver.py::ReduceAllDriverTestCase::test_merge_chunks_with_different_modality
FAILED tests/unit/drivers/querylang/test_querylang_drivers.py::QueryLangTestCase::test_filter_ql_modality
FAILED tests/unit/flow/test_flow.py::FlowTestCase::test_flow_with_modalitys_simple
```

--- a/jina/drivers/__init__.py
+++ b/jina/drivers/__init__.py
@@ -170,11 +170,11 @@ class BaseDriver(metaclass=DriverType):
 
 class BaseRecursiveDriver(BaseDriver):
 
-    def __init__(self, depth_range: Tuple[int] = (0, 0), apply_order: str = 'post',
+    def __init__(self, depth_range: Tuple[int] = (0, 1), apply_order: str = 'post',
                  traverse_on: Tuple[str] = ('chunks',), *args, **kwargs):
         """
 
-        :param depth_range: right-exclusive range of the recursion depth, (0,0) for root-level only
+        :param depth_range: right-exclusive range of the recursion depth, (0, 1) for root-level only
         :param apply_order: the traverse and apply order. if 'post' then first traverse then call apply, if 'pre' then first apply then traverse
         :param args:
         :param kwargs:
@@ -226,8 +226,10 @@ class BaseRecursiveDriver(BaseDriver):
             """
             if _docs:
                 for d in _docs:
+                    # check if apply to next level
                     if d.level_depth < self._depth_end:
                         post_traverse(getattr(d, traverse_on), traverse_on, d)
+                    # check if apply to the current level
                     if self.is_apply and (d.level_depth >= self._depth_start):
                         self._apply(d, context_doc, traverse_on, *args, **kwargs)
 
@@ -242,9 +244,11 @@ class BaseRecursiveDriver(BaseDriver):
                     self._apply_all(_docs, context_doc, traverse_on, *args, **kwargs)
 
                 for d in _docs:
+                    # check if apply on the current level
                     if self.is_apply and d.level_depth >= self._depth_start:
                         self._apply(d, context_doc, traverse_on, *args, **kwargs)
-                    if d.level_depth < self._depth_end:
+                    # check if apply to the next level
+                    if (d.level_depth + 1) < self._depth_end:
                         pre_traverse(getattr(d, traverse_on), traverse_on, d)
 
         if self.recursion_order == 'post':

--- a/jina/drivers/__init__.py
+++ b/jina/drivers/__init__.py
@@ -245,10 +245,10 @@ class BaseRecursiveDriver(BaseDriver):
 
                 for d in _docs:
                     # check if apply on the current level
-                    if self.is_apply and d.level_depth >= self._depth_start:
+                    if self.is_apply and self._depth_start <= d.level_depth < self._depth_end:
                         self._apply(d, context_doc, traverse_on, *args, **kwargs)
                     # check if apply to the next level
-                    if (d.level_depth + 1) < self._depth_end:
+                    if d.level_depth < self._depth_end:
                         pre_traverse(getattr(d, traverse_on), traverse_on, d)
 
         if self.recursion_order == 'post':

--- a/jina/drivers/__init__.py
+++ b/jina/drivers/__init__.py
@@ -230,7 +230,7 @@ class BaseRecursiveDriver(BaseDriver):
                     if d.level_depth < self._depth_end:
                         post_traverse(getattr(d, traverse_on), traverse_on, d)
                     # check if apply to the current level
-                    if self.is_apply and (d.level_depth >= self._depth_start):
+                    if self.is_apply and self._depth_start <= d.level_depth < self._depth_end:
                         self._apply(d, context_doc, traverse_on, *args, **kwargs)
 
                 # check first doc if in the required depth range

--- a/tests/unit/drivers/querylang/test_querylang_drivers.py
+++ b/tests/unit/drivers/querylang/test_querylang_drivers.py
@@ -128,7 +128,7 @@ class QueryLangTestCase(JinaTestCase):
 
         f = (Flow().add(uses='DummyModeIdSegmenter')
             .add(
-            uses='- !FilterQL | {lookups: {modality: mode2}, traverse_on: [chunks], depth_range: [1, 1]}'))
+            uses='- !FilterQL | {lookups: {modality: mode2}, traverse_on: [chunks], depth_range: [1, 2]}'))
 
         with f:
             f.index(random_docs_with_chunks(), output_fn=validate, callback_on_body=True)

--- a/tests/unit/drivers/yaml/mockencoder-mode1.yml
+++ b/tests/unit/drivers/yaml/mockencoder-mode1.yml
@@ -6,9 +6,9 @@ requests:
         with:
           lookups: {modality: mode1}
           traverse_on: [chunks]
-          depth_range: [1, 1]
+          depth_range: [1, 2]
       - !EncodeDriver
         with:
           method: encode
           traverse_on: [chunks]
-          depth_range: [1, 1]
+          depth_range: [1, 2]

--- a/tests/unit/drivers/yaml/mockencoder-mode2.yml
+++ b/tests/unit/drivers/yaml/mockencoder-mode2.yml
@@ -6,9 +6,9 @@ requests:
         with:
           lookups: {modality: mode2}
           traverse_on: [chunks]
-          depth_range: [1, 1]
+          depth_range: [1, 2]
       - !EncodeDriver
         with:
           method: encode
           traverse_on: [chunks]
-          depth_range: [1, 1]
+          depth_range: [1, 2]

--- a/tests/unit/flow/test_flow.py
+++ b/tests/unit/flow/test_flow.py
@@ -508,7 +508,7 @@ class FlowTestCase(JinaTestCase):
 
         flow = Flow().add(name='chunk_seg', parallel=3, uses='_pass').\
             add(name='encoder12', parallel=2,
-                uses='- !FilterQL | {lookups: {modality__in: [mode1, mode2]}, depth_range: [0, 0]}')
+                uses='- !FilterQL | {lookups: {modality__in: [mode1, mode2]}, depth_range: [0, 1]}')
         with flow:
             flow.index(input_fn=input_fn, output_fn=validate)
 

--- a/tests/unit/flow/yaml/mockencoder-mode1.yml
+++ b/tests/unit/flow/yaml/mockencoder-mode1.yml
@@ -6,9 +6,9 @@ requests:
         with:
           lookups: {modality__in: [mode1]}
           traverse_on: [chunks]
-          depth_range: [1, 1]
+          depth_range: [1, 2]
       - !EncodeDriver
         with:
           method: encode
           traverse_on: [chunks]
-          depth_range: [1, 1]
+          depth_range: [1, 2]

--- a/tests/unit/flow/yaml/mockencoder-mode2.yml
+++ b/tests/unit/flow/yaml/mockencoder-mode2.yml
@@ -6,9 +6,9 @@ requests:
         with:
           lookups: {modality__in: [mode2]}
           traverse_on: [chunks]
-          depth_range: [1, 1]
+          depth_range: [1, 2]
       - !EncodeDriver
         with:
           method: encode
           traverse_on: [chunks]
-          depth_range: [1, 1]
+          depth_range: [1, 2]

--- a/tests/unit/flow/yaml/numpy-indexer-1.yml
+++ b/tests/unit/flow/yaml/numpy-indexer-1.yml
@@ -19,7 +19,7 @@ requests:
     IndexRequest:
       - !VectorIndexDriver
         with:
-          depth_range: [1, 1]
+          depth_range: [1, 2]
           traverse_on: [chunks]
           executor: vecidx1
       - !ExcludeQL

--- a/tests/unit/flow/yaml/numpy-indexer-1.yml
+++ b/tests/unit/flow/yaml/numpy-indexer-1.yml
@@ -31,6 +31,7 @@ requests:
             - text
       - !KVIndexDriver
         with:
+          depth_range: [1, 2]
           executor: kvidx1
     ControlRequest:
       - !ControlReqDriver {}

--- a/tests/unit/flow/yaml/numpy-indexer-2.yml
+++ b/tests/unit/flow/yaml/numpy-indexer-2.yml
@@ -19,7 +19,7 @@ requests:
     IndexRequest:
       - !VectorIndexDriver
         with:
-          depth_range: [1, 1]
+          depth_range: [1, 2]
           traverse_on: [chunks]
           executor: vecidx2
       - !ExcludeQL

--- a/tests/unit/flow/yaml/numpy-indexer-2.yml
+++ b/tests/unit/flow/yaml/numpy-indexer-2.yml
@@ -31,6 +31,7 @@ requests:
             - text
       - !KVIndexDriver
         with:
+          depth_range: [1, 2]
           executor: kvidx2
     ControlRequest:
       - !ControlReqDriver {}

